test/config-luatest/credentials_applier_test.lua

```diff
@@ -0,0 +1,480 @@
+    }
+
+    local intermediate_custom_privileges = {
+        ['user'] = {},
+        ['role'] = {
+            ['myrole1'] = {
+                ['execute'] = true,
+            },
+            ['myrole2'] = {
+                ['execute'] = true,
+            },
+            ['public'] = {
+                ['execute'] = true,
+            },
+        },
+        ['space'] = {
+            [''] = {
+                ['read'] = true,
+                ['write'] = true,
+            },
+        },
+        ['function'] = {
+            ['myfunc1'] = {
+                ['execute'] = true,
+            },
+            ['myfunc2'] = {
+                ['execute'] = true,
+            },
+        },
+        ['sequence'] = {
+            ['myseq1'] = {
+                ['read'] = true,
+                ['write'] = true,
+            },
+            ['myseq2'] = {
+                ['read'] = true,
+                ['write'] = true,
+            },
+        },
+        ['universe'] = {
+            [''] = {
+                ['read'] = true,
+            },
+        },
+    }
+
+    t.assert_equals(internal.privileges_from_box(box_custom_privileges),
+                    intermediate_custom_privileges)
+
+    t.assert_equals(internal.privileges_from_config(config_custom_data),
+                    intermediate_custom_privileges)
+end
+
+g.test_privileges_subtract = function()
+    local target = {
+        ['user'] = {},
+        ['role'] = {},
+        ['space'] = {
+            ['myspace1'] = {
+                ['read'] = true,
+                ['write'] = true,
+            },
+            ['myspace2'] = {
+                ['read'] = true,
+            }
+        },
+        ['function'] = {
+            ['myfunc1'] = {
+                ['execute'] = true,
+            },
+        },
+        ['sequence'] = {
+            ['myseq1'] = {
+                ['read'] = true,
+                ['write'] = true,
+            }
+        },
+        ['universe'] = {},
+    }
+
+    local current = {
+        ['user'] = {},
+        ['role'] = {
+            ['myrole1'] = {
+                ['execute'] = true,
+            },
+        },
+        ['space'] = {
+            ['myspace1'] = {
+                ['read'] = true,
+            },
+        },
+        ['function'] = {
+            ['myfunc1'] = {
+                ['execute'] = true,
+            },
+        },
+        ['sequence'] = {},
+        ['universe'] = {},
+    }
+
+    local lack = {{
+            obj_type = 'space',
+            obj_name = 'myspace1',
+            privs = {'write'},
+        }, {
+            obj_type = 'space',
+            obj_name = 'myspace2',
+            privs = {'read'},
+        }, {
+            obj_type = 'sequence',
+            obj_name = 'myseq1',
+            privs = {'read', 'write'},
+        },
+    }
+
+    t.assert_items_equals(internal.privileges_subtract(target, current), lack)
+end
+
+g.test_privileges_add_defaults = function(g)
+    local cases = {
```

Comment:
I don't insist, but it looks like it can be parametrized by luatest facilities.

Reply:
I would prefer to leave it as is. There is no need for extra parallelization and the test would look a bit more complex than it needs to be.

Reply:
I agree that we have a trade-off between output beauty (plus ability to run test cases separately) and readability of the testing code. I usually don't insist on parametrization using luatest facilities for small test cases.

I'm going to accept the patch as is.